CustomVideoRenderer/VideoRenderer.cs

```diff
@@ -0,0 +1,126 @@
+﻿using OpenTok;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+
+namespace CustomVideoRenderer
+{
+    /// <summary>
+    /// Follow steps 1a or 1b and then 2 to use this custom control in a XAML file.
+    ///
+    /// Step 1a) Using this custom control in a XAML file that exists in the current project.
+    /// Add this XmlNamespace attribute to the root element of the markup file where it is 
+    /// to be used:
+    ///
+    ///     xmlns:MyNamespace="clr-namespace:CustomRendererSample"
+    ///
+    ///
+    /// Step 1b) Using this custom control in a XAML file that exists in a different project.
+    /// Add this XmlNamespace attribute to the root element of the markup file where it is 
+    /// to be used:
+    ///
+    ///     xmlns:MyNamespace="clr-namespace:CustomRendererSample;assembly=CustomRendererSample"
+    ///
+    /// You will also need to add a project reference from the project where the XAML file lives
+    /// to this project and Rebuild to avoid compilation errors:
+    ///
+    ///     Right click on the target project in the Solution Explorer and
+    ///     "Add Reference"->"Projects"->[Browse to and select this project]
+    ///
+    ///
+    /// Step 2)
+    /// Go ahead and use your control in the XAML file.
+    ///
+    ///     <MyNamespace:VideoRenderer/>
+    ///
+    /// </summary>
+    public class VideoRenderer : Control, IVideoRenderer
+    {
+        private int width = -1;
```

Comment:
@Nitrillo: All property names should use PascalCase: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions.

I think the C# naming guidelines suggest using camelCase is only used for parameter names. So in the VideoRenderer class, change `width`, `height`, and `videoBitmap` to `Width`, `Height`, and `VideoBitmap`. Also, change the `brush` variable to `Brush`. And check the naming in the other sample code. Note also that the guidelines say that even constant names should use PascalCase instead of CAPITALIZED_SNAKE_CASE (e.g., `ApiKey`, `SessionId`, and `Token`), but I read that some C# developers prefer CAPITALIZED_SNAKE_CASE.